ndk/src/font.rs

```diff
@@ -0,0 +1,457 @@
+//! Bindings for [`AFont`], [`AFontMatcher`], and [`ASystemFontIterator`]
+//!
+//! [`AFont`]: https://developer.android.com/ndk/reference/group/font
+//! [`AFontMatcher`]: https://developer.android.com/ndk/reference/group/font#afontmatcher_create
+//! [`ASystemFontIterator`]: https://developer.android.com/ndk/reference/group/font#asystemfontiterator_open
+
+#![cfg(feature = "api-level-29")]
+
+use std::convert::TryFrom;
+use std::ffi::{CStr, OsStr};
+use std::fmt::{self, Write};
+use std::os::unix::prelude::OsStrExt;
+use std::path::Path;
+use std::ptr::NonNull;
+
+/// An integer holding a valid font weight value between 1 and 1000.
+///
+/// See the following definitions for more details:
+/// * [`AFONT_WEIGHT_*`]
+/// * [`Font::weight`]
+///
+/// [`AFONT_WEIGHT_*`]: https://developer.android.com/ndk/reference/group/font#anonymous-enum-33
+#[derive(Clone, Copy, Debug, PartialEq, Eq, PartialOrd, Ord)]
+pub struct FontWeight(u16);
+
+impl FontWeight {
+    pub const fn new(value: u16) -> Result<Self, FontWeightValueError> {
+        if Self::MIN.0 <= value && value <= Self::MAX.0 {
+            Ok(Self(value))
+        } else {
+            Err(FontWeightValueError(()))
+        }
+    }
+
+    pub const fn to_u16(self) -> u16 {
+        self.0
+    }
+
+    // + 1 is to keep consistency between FontWeight and other types like std::num::NonZero*
+    pub const MIN: FontWeight = FontWeight(ffi::AFONT_WEIGHT_MIN as u16 + 1);
```

Comment:
More specifically: consistency here means that it can be used as range-inclusive?

It is indeed very strange to have `0 < x <= MAX` while we're usually doing zero-based indexing with `0 <= x < MAX`, but maybe it makes sense for font weights.

Reply:
Yes, consistency here means that the `MIN` to `MAX` range is inclusive.

I investigated several implementations in other frameworks, and it seems that FontWeight classes usually don't include constants for `MIN` or `MAX`:
* Class has a constructor with an integer parameter
  * WPF `System.Windows.FontWeight`
  * Compose `androidx.compose.ui.text.font.FontWeight`
* Restricts the user to select one of the nine members of the enum
  * JavaFX `javafx.scene.text.FontWeight`
  * UIKit `UIFontWeight`
  * Flutter `FontWEight`

Usually, `MIN` to `MAX` range is always inclusive, not only for `std::num::NonZero*`, but also similar integer types in other languages.
* C++ `std::numeric_limits<T>::min()` & `std::numeric_limits<T>::max()`
* Java `Integer.MIN_VALUE` & `Integer.MAX_VALUE`
* Swift `Int.min` & `Int.max`
* Go `math.MinInt` & `math.MaxInt`

Making this range exclusive allows constructing an integer instant with an invalid value, and I think that's why `std::num::NonZeroU*::MIN` is 1.

Reply:
I mean to explain that (what you wrote in the first sentence) in the comment (that could perhaps be a doc-comment?).